bitcoin/src/blockdata/weight.rs

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! Weight
+//!
+//! This module contains the [`Weight`] struct and related methods to operate on it
+//! Block weight represents virtual size (vsize) of a transaction measured in virtual bytes (vbytes).
+
+// ensure explicit constructor
+use std::ops::{Add, Sub, AddAssign, SubAssign, Mul};
+
+/// Represents virtual transaction size
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub struct Weight(usize);
```

Comment:
I suspect `u64` is better. Weight can be larger than actual size in memory and `u64` will better interact with `Amount` and `FeeRate`.